### ISSUE TITLE
508-defect-1 ❗[SCREENREADER]: Some non-English links have the `lang="en"` attribute

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#27619]
# [508-defect-1 :exclamation:](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a Spanish speaking screen reader user, I expect content like phone numbers and links to be read out to me in Spanish, not English.

## Details

After running a WAVE report on the prod pages, I noticed some links still have the `lang="en"` attribute which I was unable to catch in staging. For example, the phone number 800-698-2411 will be pronounced in English as "Eight, zero, zero..." instead of "Ocho, cero, cero..."

## Acceptance Criteria

- [x] All Spanish content are children of an element with the `lang="es"` attribute or have the `lang="es"` attribute
- [x] Spanish content does not have the `lang="en"` attribute

## Steps to Recreate

Run a wave audit on all the prod pages and confirm some Spanish content still has `lang="en"` attributes.
https://wave.webaim.org/report#/https://www.va.gov/asistencia-y-recursos-en-espanol/

## Proposed Solution (if known)

Using the wave audit or search tools, reconfirm that no Spanish content has the `lang="en"` attribute.

## Screenshots or Trace Logs

<img width="527" alt="Screen Shot 2021-07-20 at 12 52 32 PM" src="https://user-images.githubusercontent.com/14154792/126364352-9c26914e-4bab-4af3-9693-e9f6812a985d.png">
<img width="799" alt="Screen Shot 2021-07-20 at 12 52 40 PM" src="https://user-images.githubusercontent.com/14154792/126364353-a93e6b98-895b-4daa-909f-3a5bdc97ad38.png">
<img width="809" alt="Screen Shot 2021-07-20 at 12 52 47 PM" src="https://user-images.githubusercontent.com/14154792/126364354-6e741282-6a59-4c40-9321-862989ed12e6.png">

**Comment by adamwhitlock1** [comment]
I have a PR ready and approved for this here:
[https://github.com/department-of-veterans-affairs/vets-website/pull/18035](https://github.com/department-of-veterans-affairs/vets-website/pull/18035)

The review instances for PRs are currently broken, and since the LanguageSupport component is a global footer component I would like to verify and test a little further on a review instance, so I will hold off right now on merging it. Once the review instances are working I will get it merged asap.

**Comment by adamwhitlock1** [comment]
@joshkimux Can you review this solution and see if this covers all bases on this review instance?

http://523f083e7f47921723f53ce4c118553a.review.vetsgov-internal/health-care/covid-19-vaccine-esp/

I believe it makes the page more uniform with the language tags and is less manipulative overall, but would like any feedback on things that may require more work.

**Comment by joshkimux** [comment]
@adamwhitlock1 , I'm having trouble accessing this page with SOCKS. By any chance did it spin out over the weekend?

**Comment by adamwhitlock1** [comment]
@joshkimux This issue's PR got merged, so I'm guessing that the review instances get destroyed after a certain amount of time after the merge?

The current live urls like [covid-19-vaccine-esp](https://www.va.gov/health-care/covid-19-vaccine-esp) should show the new new code reflected in it for further review.

**Comment by joshkimux** [comment]
Only one more `lang` related issue is here I believe in the breadcrumb. Adding `lang="es"` as an attribute should fix the issue.
<img width="1455" alt="Screen Shot 2021-08-02 at 4 36 11 PM" src="https://user-images.githubusercontent.com/14154792/127920684-0902b2cc-512e-4fba-a502-51db6e8f3e14.png">

On another note-- I caught some headings issues with the page that I didn't detect earlier.
<img width="1437" alt="Screen Shot 2021-08-02 at 2 54 56 PM" src="https://user-images.githubusercontent.com/14154792/127920821-22d4d1e7-d86f-40bc-a5c9-8bc4435d330b.png">

It looks like the question headings are `h2`s when they should be one level below their parent, in this case `h4`s.

**Comment by adamwhitlock1** [comment]
@joshkimux, that is a great point on the breadcrumbs! I will work on getting a PR in to fix that.

> It looks like the question headings are `h2`s when they should be one level below their parent, in this case `h4`s.

I'm wondering what the process would be for updating the va-accordion for the h2s to h4s since they are a global design system component and their implementation is not unique to these pages in question?

**Comment by joshkimux** [comment]
@adamwhitlock1 great question-- from my goldfish memory, I was almost certain there was a way to do so relatively easily without having to edit the global system component. @Mottie could you keep me honest here? 🙏

**Comment by SKasimow** [comment]
Adding in @DanielleThierryUSDSVA as she may have insight on accordion component question above!

**Comment by adamwhitlock1** [comment]
I found the `va-accordion` code in the component library that does indicate that a prop called `level` can be passed to it: [va-accordion-item.tsx](https://github.com/department-of-veterans-affairs/component-library/blob/4ee6bb1b0dfe33f029d13b8ec5cd10b61bffdcb3/src/components/va-accordion/va-accordion-item.tsx#L36)

Unfortunately that prop isn't provided for the instance of `va-accordion` in the drupal liquid template: [q_a.collapsible_panel.drupal.liquid](https://github.com/department-of-veterans-affairs/content-build/blob/f0eb489a66c155d603bdc7e9671e45082929043e/src/site/paragraphs/q_a.collapsible_panel.drupal.liquid#L9) and so it is using the default level of h2.

Thinking about this `q_a.collapsible_panel` code more, I don't know that there is a way for us to really detect what the heading level should be in regards to the previous heading / overall page hierarchy as it is built right now. For this particular page in question, the previous heading is provided via a wysiwyg content block, which means it _could_ have easily been any level of heading depending on the context around it.

I'm probably overthinking it, but maybe the `q_a.collapsible_panel` would need to have a field in the drupal cms UI to specify what level of heading should be used so that it could be customized to fall in line with the page hierarchy on a page by page basis?

Something like this perhaps:
![Screen Shot 2021-08-03 at 2 48 53 PM](https://user-images.githubusercontent.com/8332986/128084780-1e934365-9537-47bc-bcee-1a6466c31342.png)

I'm not sure that would be a viable solution though. Thoughts?

**Comment by joshkimux** [comment]
> I found the `va-accordion` code in the component library that does indicate that a prop called `level` can be passed to it: [va-accordion-item.tsx](https://github.com/department-of-veterans-affairs/component-library/blob/4ee6bb1b0dfe33f029d13b8ec5cd10b61bffdcb3/src/components/va-accordion/va-accordion-item.tsx#L36)
> 
> Unfortunately that prop isn't provided for the instance of `va-accordion` in the drupal liquid template: [q_a.collapsible_panel.drupal.liquid](https://github.com/department-of-veterans-affairs/content-build/blob/f0eb489a66c155d603bdc7e9671e45082929043e/src/site/paragraphs/q_a.collapsible_panel.drupal.liquid#L9) and so it is using the default level of h2.
> 
> Thinking about this `q_a.collapsible_panel` code more, I don't know that there is a way for us to really detect what the heading level should be in regards to the previous heading / overall page hierarchy as it is built right now. For this particular page in question, the previous heading is provided via a wysiwyg content block, which means it _could_ have easily been any level of heading depending on the context around it.
> 
> I'm probably overthinking it, but maybe the `q_a.collapsible_panel` would need to have a field in the drupal cms UI to specify what level of heading should be used so that it could be customized to fall in line with the page hierarchy on a page by page basis?
> 
> Something like this perhaps:
> ![Screen Shot 2021-08-03 at 2 48 53 PM](https://user-images.githubusercontent.com/8332986/128084780-1e934365-9537-47bc-bcee-1a6466c31342.png)
> 
> I'm not sure that would be a viable solution though. Thoughts?

Any way to let a human being specify the heading level on a case by case basis would be great!

**Comment by RLHecht** [comment]
The way the template is set up currently in Drupal, a content editor can't change the heading level for accordions. This would need to be something fixed in the front end by a developer on the public websites team. @kelsonic @brianalloyd

**Comment by brianalloyd** [comment]
PW team will review during team grooming and scrub into issue.

**Comment by kelsonic** [comment]
The new accordion web component that @Rleahy22 recently implemented actually auto-calculates what header to use afaik. Assuming it is already being used, the headers should calculate themselves 🙂

**Comment by adamwhitlock1** [comment]
@joshkimux Do you think we can close this out?

Work to fix the breadcrumb lang attr was merged here:
https://app.zenhub.com/workspace/o/department-of-veterans-affairs/vets-website/issues/18140

**Comment by joshkimux** [comment]
Yes, but could we open up a new ticket for the accordion headings?

**Comment by adamwhitlock1** [comment]
Yea, I think that is a good idea. @Rleahy22 Do you know if the new va-accordion web component is supposed to auto-calculate the heading level that it uses? I couldn't find any code that would indicate that.

**Comment by Rleahy22** [comment]
It does not auto-calculate as far as I can tell. It defaults to `2`
https://github.com/department-of-veterans-affairs/component-library/blob/web-components/src/components/va-accordion/va-accordion-item.tsx#L36
